src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -114,13 +119,24 @@ public function setIgnoredAttributes(array $ignoredAttributes)
     /**
      * Set attributes to be camelized on denormalize.
      *
+     * @deprecated Deprecated since version 2.7, to be removed in 3.0. Use Symfony\Component\Serializer\NameConverter\CamelCaseToSnakeCaseNameConverter instead.
+     *
      * @param array $camelizedAttributes
      *
      * @return self
      */
     public function setCamelizedAttributes(array $camelizedAttributes)
     {
-        $this->camelizedAttributes = $camelizedAttributes;
+        trigger_error(sprintf('%s is deprecated since version 2.7 and will be removed in 3.0. Use Symfony\Component\Serializer\NameConverter\CamelCaseToSnakeCaseNameConverter instead.', __METHOD__), E_USER_DEPRECATED);
+
+        $attributes = array();
+        foreach ($camelizedAttributes as $camelizedAttribute) {
+            $attributes[] = lcfirst(preg_replace_callback('/(^|_|\.)+(.)/', function ($match) {
+                return ('.' === $match[1] ? '_' : '').strtoupper($match[2]);
+            }, $camelizedAttribute));
+        }
+
+        $this->nameConverter = new CamelCaseToSnakeCaseNameConverter($attributes);
```

Comment:
this will break if a NameConverter was passed in the constructor (you will removed the existing one entirely)